lua/compe/matcher.lua

```diff
@@ -15,9 +15,7 @@ Matcher.match = function(context, source, items)
     local word = item.original_word
     if #input > 0 then
       if item.filter_text and #item.filter_text > 0 then
-        if Character.match(string.byte(input, 1), string.byte(item.filter_text, 1)) then
-          word = item.filter_text
-        end
+        word = item.filter_text
       end
```

Comment:
I think we should respect `prefix matched abbr` as first priority.
So I'd create another PR (#392). Please check it.